binary_search_tree/tree.py

```diff
@@ -1,59 +1,117 @@
 class TreeNode:
-    def __init__(self, key, val = None):
-        if val == None:
-            val = key
+    def __init__(self, key, value=None):
+        if value == None:
+            value = key
 
         self.key = key
-        self.value = val
+        self.value = value
         self.left = None
         self.right = None
-        
 
 
 class Tree:
     def __init__(self):
         self.root = None
 
-    # Time Complexity: 
-    # Space Complexity: 
-    def add(self, key, value = None):
-        pass
+    # Time Complexity: O(log(n))*assumes all balanced tree
+    # Space Complexity: O(1)
+    def add(self, key, value=None):
```

Comment:
👍  However due to the recursive call stack this is O(log n) for space complexity (if the tree is balanced and O(n) otherwise).